backend/qemu.pm

```diff
@@ -534,72 +508,42 @@ sub load_snapshot {
 
 sub do_extract_assets {
     my ($self, $args) = @_;
-    my $pattern;
     my $name    = $args->{name};
     my $img_dir = $args->{dir};
-
-    if ($args->{pflash_vars}) {
-        $pattern = qr/^pflash-vars$/;
-    }
-    else {
-        my $hdd_num = $args->{hdd_num} - 1;
-        $pattern = qr/^hd$hdd_num$/;
-    }
-
-    unless ($self->{proc}->has_state()) {
-        $self->{proc}->load_state();
-    }
-
+    my $hdd_num = $args->{hdd_num} - 1;
+    my $pattern = $args->{pflash_vars} ? qr/^pflash-vars$/ : qr/^hd$hdd_num$/;
+    $self->{proc}->load_state() unless $self->{proc}->has_state();
     mkpath($img_dir);
     bmwqemu::fctinfo("Extracting $pattern");
     my $res = $self->{proc}->export_blockdev_images($pattern, $img_dir, $name);
     die "Expected one drive to be exported, not $res" if $res != 1;
 }
 
-
 # baseclass virt method overwrite end
 
-sub find_ovmf {
-    foreach my $firmware (@bmwqemu::ovmf_locations) {
-        return $firmware if -e $firmware;
-    }
-}
+sub find_ovmf { first { -e } @bmwqemu::ovmf_locations }
 
-sub get_virtio_console_names {
-    my $self = shift;
-    my @names;
-    if ($bmwqemu::vars{VIRTIO_CONSOLE}) {
-        for (my $i = 0; $i < ($bmwqemu::vars{VIRTIO_CONSOLE_NUM} // 1); $i++) {
-            my $name = 'virtio_console' . ($i ? $i : '');
-            push(@names, $name);
-        }
-    }
-    return @names;
+sub virtio_console_names {
+    return () unless $bmwqemu::vars{VIRTIO_CONSOLE};
+    return map { 'virtio_console' . ($_ || '') } (0 .. ($bmwqemu::vars{VIRTIO_CONSOLE_NUM} // 1));
 }
 
-sub get_virtio_console_fifo_names {
-    my $self = shift;
-    return map { ($_ . '.in', $_ . '.out') } $self->get_virtio_console_names();
+sub virtio_console_fifo_names { map { $_ . '.in', $_ . '.out' } virtio_console_names }
+
+sub console_fifo {
+    my ($name) = @_;
+    return bmwqemu::fctwarn("Fifo pipe '$name' already exists!") if -e $name;
+    mkfifo($name, 0666) or bmwqemu::fctwarn("Failed to create pipe $name: $!");
 }
 
 sub create_virtio_console_fifo {
     my $self = shift;
```

Comment:
Looks like you can remove this line here as well.

Reply:
done